src/server/api/routers/application.ts

```diff
@@ -37,7 +37,13 @@ export const applicationRouter = createTRPCRouter({
 
   save: protectedProcedure
     .input(applicationSaveSchema)
-    .mutation(({ input, ctx }) => {
-      // TODO: handler body
+    .mutation( async ({ input, ctx }) => {
+      const userId = ctx.session.user.id;
+      const applicationData = input;
+  
+      const savedApplication = await ctx.db.insert(applications).values({...applicationData, userId})
+        .onConflictDoUpdate({target: applications.id, set: {...applicationData, userId}})
```

Comment:
```suggestion
      const savedApplication = await ctx.db.insert(applications).values({...applicationData, userId})
        .onConflictDoUpdate({target: [applications.id, applications.userId], set: {...applicationData, userId}})
```

I think we should add `userId` as a target so that it only updates an application made by the user. We know that the user's ID is correct since we do an auth check (it's a protected query), but we don't necessarily know if the application.id is owned by the user without this.